packages/dev/core/src/Materials/material.ts

```diff
@@ -1896,7 +1896,7 @@ export class Material implements IAnimatable, IClipPlanesHolder {
                     for (const subMesh of mesh.subMeshes) {
                         geometry._releaseVertexArrayObject(subMesh.effect);
                         if (forceDisposeEffect && subMesh.effect) {
-                            subMesh.effect.dispose();
+                            subMesh._drawWrapper.dispose(true);

```

Comment:
I don't think this is the right fix.

It makes the PG work because `DrawWrapper.effect` is reset to `null`, so the second time we run `DrawWrapper.dispose` (because of `mesh.material = null`), we won't call `effect.dispose()` again. But it's a bug that `DrawWrapper.dispose` is called a second time on an already disposed draw wrapper => we should remove the `drawWrapper` instance from `subMesh._drawWrappers` to avoid this.

So, doing `subMesh._removeDrawWrapper(engine.currentRenderPassId, true, true);` instead of `subMesh._drawWrapper.dispose(true)` would be better, but it's still not ok.

What we want is to dispose the draw wrappers that are related to the material we are currently releasing, that is the draw wrappers that have been created by this material. So, we should loop over `subMesh._drawWrappers` and only remove the right ones. The problem is that we currently can't relate a `DrawWrapper` to a material...

I think we should talk about this problem and what we can do about it.

Reply:
While I agree with you on a philosophical level, the current solution is fast and reliable.

Until a better solution is found we have to keep this one